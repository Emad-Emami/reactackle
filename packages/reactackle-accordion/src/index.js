import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
