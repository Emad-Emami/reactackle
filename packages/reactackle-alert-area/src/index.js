import AlertArea from './AlertArea';

export { AlertArea };
