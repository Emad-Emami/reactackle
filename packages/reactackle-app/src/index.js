import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
