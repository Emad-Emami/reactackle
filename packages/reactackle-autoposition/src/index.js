import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
