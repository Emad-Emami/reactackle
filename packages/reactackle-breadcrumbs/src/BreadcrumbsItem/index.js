import BreadcrumbsItem from './BreadcrumbsItem';

export { BreadcrumbsItem };
