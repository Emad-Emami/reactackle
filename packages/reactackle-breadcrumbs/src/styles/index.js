import componentTheme from './theme';

export { componentTheme };

export * from './BreadcrumbsStyled';
export * from './ListWrapperStyled';
export * from './ListStyled';
export * from './NavigationButtonStyled';
