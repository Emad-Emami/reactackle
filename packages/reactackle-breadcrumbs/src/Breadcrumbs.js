import React, { Component } from 'react';
import PropTypes from 'prop-types';
import throttle from 'lodash.throttle';
import { Button } from 'reactackle-button';
import { noop, getKey, registerDefaultComponentTheme } from 'reactackle-core';
import { IconArrowChevronRight, IconArrowChevronLeft } from 'reactackle-icons';
import { BreadcrumbsItem } from './BreadcrumbsItem';

import {
  componentTheme,
  BreadcrumbsStyled,
  ListWrapperStyled,
  ListStyled,
  NavigationButtonStyled,
} from './styles';

registerDefaultComponentTheme('breadcrumbs', componentTheme);

const propTypes = {
  /**
   * Array of breadcrumbs items
   */
  items: PropTypes.arrayOf(PropTypes.shape(BreadcrumbsItem.propTypes))
    .isRequired,
  /**
   * If true, breadcrumbs will become scrollable to fit wrapper width
   */
  overflow: PropTypes.bool,
  /**
   * Component to be used as a link in BreadcrumbItem
   */
  linkComponent: PropTypes.func,
  /**
   * Define component's color scheme
   */
  colorScheme: PropTypes.oneOf(['light', 'dark']),
  /**
   * Vertical align of items
   */
  alignItems: PropTypes.oneOf(['top', 'center', 'bottom']),
  /**
   * Type of separator between items
   */
  separatorType: PropTypes.oneOf(['text', 'icon']),
  /**
   * Define separator
   */
  separatorIcon: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
  ]),
  /**
   * Function to be used on item's click
   */
  onItemClick: PropTypes.func,
};
const defaultProps = {
  linkComponent: null,
  overflow: false,
  colorScheme: 'dark',
  alignItems: 'center',
  separatorType: 'icon',
  separatorIcon: <IconArrowChevronRight size="custom" color="inherit" />,
  onItemClick: noop,
};

export default class Breadcrumbs extends Component {
  constructor(props) {
    super(props);

    this.itemDomNodes = new Map();
    this.wrapperDomNode = null;
    this.lastId = props.items.length - 1;
    this.scrollAfterRender = false;
    this.animateScroll = false;
    this.animateEnd = true;

    this.state = {
      showLeftButton: false,
      showRightButton: false,
      offset: 0,
      id: 0,
    };

    this._handleNext = this._handleNext.bind(this);
    this._handlePrev = this._handlePrev.bind(this);
    this._handleTransitionEnd = this._handleTransitionEnd.bind(this);
    this._handleResize = throttle(this._handleResize.bind(this), 100);
    this._saveWrapListRef = this._saveWrapListRef.bind(this);
    this._saveItemRef = this._saveItemRef.bind(this);
    this._saveContainerRef = this._saveContainerRef.bind(this);
  }

  componentDidMount() {
    this.containerDomNode.addEventListener('transitionend',
      this._handleTransitionEnd, false);
    this._showButtonsAndScroll(this.lastId);
    window.addEventListener('resize', this._handleResize, false);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.items !== this.props.items) {
      this.lastId = nextProps.items.length - 1;

      this.setState({
        showLeftButton: false,
        showRightButton: false,
        offset: 0,
        id: 0,
      });
    }
  }

  componentDidUpdate(prevProps, prevState) {
    if (prevProps.items !== this.props.items) {
      this._showButtonsAndScroll(this.lastId);
      this.animateScroll = false;
    } else if (this.props.overflow && this.scrollAfterRender) {
      this._scrollToElement(this.state.id, this.state.id >= prevState.id);
      this.scrollAfterRender = false;
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this._handleResize, false);
    this.containerDomNode.removeEventListener('transitionend',
            this._handleTransitionEnd, false);
  }

  _isOverflow() {
    if (!this.props.overflow || this.itemDomNodes.size < 2) return false;

    const lastItem = this.itemDomNodes.get(this.lastId),
      firstItem = this.itemDomNodes.get(0),
      lastItemSize = lastItem.getBoundingClientRect(),
      firstItemSize = firstItem.getBoundingClientRect(),
      wrapListSize = this.wrapperDomNode.getBoundingClientRect(),
      overflowToRight = lastItemSize.right > wrapListSize.right,
      overflowToLeft = firstItemSize.left < wrapListSize.left;

    return overflowToRight || overflowToLeft;
  }

  _handleTransitionEnd() {
    this.animateEnd = true;
  }

  _handleResize() {
    this.animateScroll = false;
    this._scrollToCurrent();
  }

  _showButtonsAndScroll(id) {
    const newState = {
      showLeftButton: false,
      showRightButton: false,
      id,
    };

    if (this._isOverflow()) {
      if (id === 0) {
        newState.showRightButton = true;
        newState.showLeftButton = false;
      }

      if (id === this.lastId) {
        newState.showRightButton = false;
        newState.showLeftButton = true;
      }

      if (id > 0 && id < this.lastId) {
        newState.showRightButton = true;
        newState.showLeftButton = true;
      }

      this.scrollAfterRender = true;
    }

    this.setState(newState);
  }

  _saveItemRef(index, ref) {
    if (ref)
      this.itemDomNodes.set(index, ref);
    else
      this.itemDomNodes.delete(index);
  }

  _saveWrapListRef(ref) {
    this.wrapperDomNode = ref;
  }

  _saveContainerRef(ref) {
    this.containerDomNode = ref;
  }

  _scrollToElement(id, alignRight) {
    if (this.itemDomNodes.size && this.itemDomNodes.get(id)) {
      const elSize = this.itemDomNodes.get(id).getBoundingClientRect(),
        wrapperSize = this.wrapperDomNode.getBoundingClientRect();
  
      let offset = this.state.offset + elSize.left - wrapperSize.left;
  
      if (alignRight)
        offset -= (wrapperSize.width - elSize.width);
  
      this.setState({
        offset,
      });
    }
  }

  _scrollToCurrent() {
    this._showButtonsAndScroll(this.state.id);
  }

  _handleNext() {
    if (!this.animateEnd) return;

    const wrapListSize = this.wrapperDomNode.getBoundingClientRect();
    let newId = this.state.id;

    while (newId < this.lastId) {
      newId += 1;

      if (!this.itemDomNodes.has(newId)) break;

      const newItem = this.itemDomNodes.get(newId),
        newItemSize = newItem.getBoundingClientRect();

      if (newItemSize.right > wrapListSize.right) {
        this._showButtonsAndScroll(newId);
        break;
      }
    }

    this.animateScroll = true;
    this.animateEnd = false;
  }

  _handlePrev() {
    if (!this.animateEnd) return;

    const wrapListSize = this.wrapperDomNode.getBoundingClientRect();
    let newId = this.state.id;

    while (newId > 0) {
      newId -= 1;

      if (!this.itemDomNodes.has(newId)) break;

      const newItem = this.itemDomNodes.get(newId),
        newItemSize = newItem.getBoundingClientRect();

      if (newItemSize.left < wrapListSize.left) {
        this._showButtonsAndScroll(newId);
        break;
      }
    }

    this.animateScroll = true;
    this.animateEnd = false;
  }

  _renderLeftButton() {
    if (!this.props.overflow || !this.state.showLeftButton) return null;

    return (
      <NavigationButtonStyled left>
        <Button
          icon={<IconArrowChevronLeft size="custom" color="inherit" />}
          onPress={this._handlePrev}
          colorScheme={this.props.colorScheme === 'light'
            ? 'flatLight'
            : 'flat'}
        />
      </NavigationButtonStyled>
    );
  }

  _renderRightButton() {
    if (!this.props.overflow || !this.state.showRightButton) return null;

    return (
      <NavigationButtonStyled right>
        <Button
          icon={<IconArrowChevronRight size="custom" color="inherit" />}
          onPress={this._handleNext}
          colorScheme={this.props.colorScheme === 'light'
            ? 'flatLight'
            : 'flat'
          }
        />
      </NavigationButtonStyled>
    );
  }

  render() {
    const {
        separatorType,
        separatorIcon,
        linkComponent,
        onItemClick,
        colorScheme,
        alignItems,
        items,
    } = this.props;
    
    const BreadcrumbsItemsList = items.map((item, index, array) => (
      <BreadcrumbsItem
        index={index}
        title={item.title}
        subtitle={item.subtitle}
        separatorType={separatorType}
        separatorIcon={separatorIcon}
        home={item.home}
        itemHref={item.itemHref}
        isActive={index === array.length - 1}
        linkComponent={linkComponent}
        onClick={onItemClick}
        saveRef={this._saveItemRef}
        options={item.options}
        key={getKey(item, index)}
        colorScheme={colorScheme}
        {...item}
      />
    ));

    const style = {
      transform: `translateX(-${this.state.offset}px)`,
    };

    if (this.animateScroll)
      style.transition = '0.3s ease-in-out';

    return (
      <BreadcrumbsStyled aria-label="Breadcrumb">
        {this._renderLeftButton()}

        <ListWrapperStyled innerRef={this._saveWrapListRef}>
          <ListStyled
            style={style}
            alignItems={alignItems}
            innerRef={this._saveContainerRef}
          >
            {BreadcrumbsItemsList}
          </ListStyled>
        </ListWrapperStyled>

        {this._renderRightButton()}
      </BreadcrumbsStyled>
    );
  }
}

Breadcrumbs.propTypes = propTypes;
Breadcrumbs.defaultProps = defaultProps;
