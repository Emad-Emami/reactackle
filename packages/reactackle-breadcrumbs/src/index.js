import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
