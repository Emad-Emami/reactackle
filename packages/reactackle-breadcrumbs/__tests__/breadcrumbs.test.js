import React from 'react';
import renderer from 'react-test-renderer';
import { IconSvg } from 'reactackle-icons';
import { Breadcrumbs } from '../src';

jest.mock('react-dom');

const rendererOptions = {
  createNodeMock() {
    return {
      addEventListener() {},
    };
  },
};

const items = [
  {
    title: 'Index',
    home: true,
    itemHref: 'index',
  },
];

//eslint-disable-next-line
const CustomLink = props => <a {...props} />;

describe('<Breadcrumbs />', () => {
  it('must render with default props', () => {
    const component = renderer.create(
      <Breadcrumbs items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with alignItems prop set to one from enum', () => {
    const component = renderer.create(
      <Breadcrumbs alignItems="top" items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with colorScheme prop set to one from enum', () => {
    const component = renderer.create(
      <Breadcrumbs colorScheme="light" items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with overflow prop set to true', () => {
    const component = renderer.create(
      <Breadcrumbs overflow items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with separatorType prop set to text', () => {
    const component = renderer.create(
      <Breadcrumbs separatorType="text" items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with separatorIcon prop set to <IconSvg />', () => {
    const component = renderer.create(
      <Breadcrumbs separatorIcon={<IconSvg />} items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('must render with linkComponent prop set to CustomLink', () => {
    const component = renderer.create(
      <Breadcrumbs linkComponent={CustomLink} items={items} />,
      rendererOptions,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

});
