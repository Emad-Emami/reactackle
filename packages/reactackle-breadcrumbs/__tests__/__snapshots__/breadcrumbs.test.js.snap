// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumbs /> must render with alignItems prop set to one from enum 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx kCEXgW"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with colorScheme prop set to one from enum 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX cODncM"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy fWhbNv"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with default props 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with linkComponent prop set to CustomLink 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          active={true}
          alignItems="center"
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          colorScheme="dark"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with overflow prop set to true 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with separatorIcon prop set to <IconSvg /> 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl iSGIaX"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              preserveAspectRatio="xMidYMid meet"
              viewBox="0 0 24 24"
            >
              <path
                d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;

exports[`<Breadcrumbs /> must render with separatorType prop set to text 1`] = `
<nav
  aria-label="Breadcrumb"
  className="BreadcrumbsStyled-kglnqC hcwflO"
>
  <div
    className="ListWrapperStyled-bXqrLk fUUijE"
  >
    <ol
      className="ListStyled-eVNKkx jWoxFL"
      style={
        Object {
          "transform": "translateX(-0px)",
        }
      }
    >
      <li
        className="BreadcrumbsItemStyled-gqfHBX exhQAI"
      >
        <a
          aria-current="item"
          className="AnchorStyled-cEwrfy cliYLp"
          href={null}
          onClick={[Function]}
        >
          <div
            className="IconHomeStyled-bfqTlj bEwNNP"
          >
            <div
              className="IconSvgStyled-cegYAl fddBCv"
              color=""
              rotate={0}
              size="normal"
            >
              <svg
                viewBox="0 0 24 24"
              >
                <path
                  d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                />
              </svg>
            </div>
          </div>
          <div
            className="TextBoxStyled-imlyyE izNZWH"
          >
            <div
              className="TitleStyled-kOmUSD kkUgKJ"
            >
              Index
            </div>
            
          </div>
        </a>
        <div
          className="SeparatorStyled-ksvBLD bTxELE"
        >
          <div
            className="IconSvgStyled-cegYAl fAPHID"
            color=""
            rotate={0}
            size="normal"
          >
            <svg
              viewBox="0 0 24 24"
            >
              <polygon
                points="9.7,5.8 8.3,7.2 13.1,12 8.3,16.8 9.7,18.2 15.9,12"
              />
            </svg>
          </div>
        </div>
      </li>
    </ol>
  </div>
</nav>
`;
